### PR TITLE
Kong version bump to 0.14.1

### PR DESCRIPTION
**Pull request by shashiranjan84**
**What this PR does / why we need it**:

Kong version bump to 0.14.1

### CODE REVIEW

**Comment by shashiranjan84** [comment]
@unguiculus any clue why build failing?

```
W0828 17:38:35.446] ERROR: Chart installation failed: stable/kong
W0828 17:38:35.530] Traceback (most recent call last):
W0828 17:38:35.530]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 50, in <module>
W0828 17:38:35.530]     main(ARGS.env, ARGS.cmd + ARGS.args)
W0828 17:38:35.531]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 41, in main
W0828 17:38:35.531]     check(*cmd)
W0828 17:38:35.531]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 30, in check
W0828 17:38:35.531]     subprocess.check_call(cmd)
W0828 17:38:35.531]   File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
W0828 17:38:35.531]     raise CalledProcessError(retcode, cmd)
W0828 17:38:35.531] subprocess.CalledProcessError: Command '('./test/e2e.sh',)' returned non-zero exit status 1
I0828 17:38:35.536] 0084c68d879c7a04caa4fcc55bf40d801660e1814e1d5c15947830af43eb3ee9
E0828 17:38:35.537] Command failed
I0828 17:38:35.537] process 778 exited with code 1 after 5.4m
```

**Comment by davidkarlsen** [comment]
Maybe the test passed falsely before and the problem was always there?
There seems to be some snag with a missing role: "FATAL:  role "kong" does not exist"

**Comment by shashiranjan84** [comment]
That doesn't seems to be the case as the Postgres pod come up fine 

```
I0827 21:11:04.982] You can change this by editing pg_hba.conf or using the option -A, or
I0827 21:11:04.982] --auth-local and --auth-host, the next time you run initdb.
I0827 21:11:04.982] waiting for server to start....LOG:  database system was shut down at 2018-08-27 21:06:39 UTC
I0827 21:11:04.982] LOG:  MultiXact member wraparound protections are now enabled
I0827 21:11:04.982] LOG:  database system is ready to accept connections
I0827 21:11:04.982] LOG:  autovacuum launcher started
I0827 21:11:04.982] FATAL:  role "kong" does not exist
I0827 21:11:04.983]  done
I0827 21:11:04.983] server started
I0827 21:11:04.983] CREATE DATABASE
I0827 21:11:04.983] 
I0827 21:11:04.983] CREATE ROLE
I0827 21:11:04.983] 
I0827 21:11:04.983] 
I0827 21:11:04.983] /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
I0827 21:11:04.983] 
I0827 21:11:04.983] waiting for server to shut down....LOG:  received fast shutdown request
I0827 21:11:04.983] LOG:  aborting any active transactions
I0827 21:11:04.983] LOG:  autovacuum launcher shutting down
I0827 21:11:04.984] LOG:  shutting down
I0827 21:11:04.984] LOG:  database system is shut down
I0827 21:11:04.984]  done
I0827 21:11:04.984] server stopped
I0827 21:11:04.984] 
I0827 21:11:04.984] PostgreSQL init process complete; ready for start up.
I0827 21:11:04.984] 
I0827 21:11:04.984] LOG:  database system was shut down at 2018-08-27 21:06:41 UTC
I0827 21:11:04.984] LOG:  MultiXact member wraparound protections are now enabled
I0827 21:11:04.984] LOG:  database system is ready to accept connections
I0827 21:11:04.985] LOG:  autovacuum launcher started
```

**Comment by shashiranjan84** [comment]
I am finding it difficult to figure out the problem from log, here is last successful build log https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/charts/6602/pull-charts-e2e/11594/?log#log which has same errors but build passes eventually.

**Comment by cpanato** [comment]
i will try to debug that tomorrow

**Comment by cpanato** [comment]
what is the timeout in the charts e2e testing?
I installed in my test cluster the main pod took a long time to reach the running state(more than 5 minutes).
And I think this is: because it needs to run the migration job first before the kong properly start. (I'm assuming that I'm not sure maybe @shashiranjan84 can confirm)

I would suggest adding an initcontainer in the kong pod and to wait for the migration finish and then reduce the initialDelaySeconds in the probes I don't know if that will fix the e2e tests because the migration can take longer...

Also, I would suggest adding an initcontainer in the job and in the main to wait for the Postgres be up and running

this will avoid the restarts.

hope that helps, just my two cents

**Comment by cpanato** [comment]
Another comment, since this chart is in stable I would suggest to bump the version to 1.0.0 at least

**Comment by shashiranjan84** [comment]
> what is the timeout in the charts e2e testing?
I installed in my test cluster the main pod took a long time to reach the running state(more than 5 minutes).
And I think this is: because it needs to run the migration job first before the kong properly start. (I'm assuming that I'm not sure maybe @shashiranjan84 can confirm)

Yes, that is how it was developed right from beginning. Seems like we changes something in e2e test which make 

> I would suggest adding an initcontainer in the kong pod and to wait for the migration finish and then reduce the initialDelaySeconds in the probes I don't know if that will fix the e2e tests because the migration can take longer...

Are you suggesting, curling the the migration job status in init-container? I think this is more of a hack. We should rely on K8s concept on reaching desired state.

**Comment by cpanato** [comment]
@shashiranjan84 just to understand how this app works, the migration job need to run before the main app starts correct? 

I was thinking, you right will be like a hack to check the job migration, please just ignore.
But I would add a check for the database.

**Comment by shashiranjan84** [comment]
@cpanato Yes migration supposed run before Kong to add Kong required resources in DB. Same time it need to be run from only one Kong node. So currently `migrations.yaml` deployment file start one Kong node and run the migration, it will keep trying till it run the migration successfully. Meanwhile Kong(deployment.yaml)  will keep tying to reach healthy state.

**Comment by cpanato** [comment]
@shashiranjan84 thanks for the clarification.

**Comment by shashiranjan84** [comment]
@cpanato I have added init-container to wait till DB is ready. need /ok-to-test, thanks

**Comment by shashiranjan84** [comment]
ping @davidkarlsen.

**Comment by shashiranjan84** [comment]
@davidkarlsen @unguiculus @cpanato I would need some help to resolve this issue? I did made some changes to stack cluster faster than  before but it still fails. It deployment needs to be healthy is 1st go? or there is a timeout?

File: stable/kong/templates/deployment.yaml [hunk 1]
```diff
@@ -23,6 +23,10 @@ spec:
         app: {{ template "kong.name" . }}
         release: {{ .Release.Name }}
     spec:
+      initContainers:
+      - name: init-kong-database
+        image: busybox
+        command: ['sh', '-c', 'until nslookup {{ template "kong.postgresql.fullname" . }}; do echo waiting for {{ template "kong.postgresql.fullname" . }}; sleep 2; done;']
```
Comment (hbagdi) [inline stable/kong/templates/deployment.yaml]: This checks if the pg container is up and not if the Postgres is actually accepting connections, which means this check would work fine most of the time but not always.
A more stricter check could be: `nc -zv $KONG_PG_HOST ${KONG_PG_PORT-5432};`

**Comment by hbagdi** [comment]
This seemed to have stalled.

@davidkarlsen  @cpanato @unguiculus Is there a path forward on this issue, if there has been a change in test-infra could someone rerun it?

**Comment by cpanato** [comment]
Installed locally in my k8s cluster and it works.
The initContainer for the main pod take some time and I guess this is hitting the timeout in the Chart Testing cluster

I tested several times and sometimes the init is a bit faster and other times take some time.


how should we proceed @davidkarlsen @mattfarina?
----
My kubectl get po

```
~ at ☸️ mm-test-helm took 22s
$ kubectl get po
NAME                                          READY   STATUS      RESTARTS   AGE
unhinged-sponge-kong-6d5d8fcff-gtqq9          1/1     Running     0          7m40s
unhinged-sponge-kong-migrations-84d5r         0/1     Completed   0          7m40s
unhinged-sponge-postgresql-8456455978-ns6zz   1/1     Running     0          7m40s
```

**Comment by unguiculus** [comment]
The issue might be cause by the identical label selector on the deployment and the job. I added component labels. Let's see if this helps.

**Comment by hbagdi** [comment]
@unguiculus That worked! Thank you!

**Comment by unguiculus** [comment]
Yay. Fixed. 

@shashiranjan84 The labels are necessary so CI isn't confused. It was checking the job's pod for readiness which had to fail. I'll try and fix this in the testing tool.

Before we merge this, feel free to give the labels different values if you like.

**Comment by cpanato** [comment]
/lgtm
thanks for the update @unguiculus

**Pull request merged**